prov/sockets/include/sock.h

```diff
@@ -870,7 +870,12 @@ struct sock_conn_hdr {
 	uint8_t reserved[3];
 	uint16_t port;
 	uint16_t cm_data_sz;
-	char cm_data[0];
+	/* Dear developer, this value is commented due to
+	 * issues with building on non-GCC compilers
+	 * (MS Visual Studio Compiler). Please, do not
+	 * uncomment it until you really need this value.
+	 * Thank you. */
+	/*char cm_data[0];*/
```

Comment:
I tested with cm_data[] - it doesn't work :(
as I can see this is standard violation to use zero-based array on non-last element in structure (header is used inside another structure)